### PR TITLE
Flow/v1

### PR DESCRIPTION
**Pull request by cryogenian**
[ch13130] [ch13073] 

🛑 because https://github.com/precog/quasar-lib-jdbc/pull/108

Specs are the same as for teradata plugin.

### CODE REVIEW

File: destination/src/main/scala/quasar/plugin/hana/destination/HANADestination.scala [hunk 1]
```diff
@@ -21,41 +21,60 @@ import scala._, Predef._
...
 
 import org.slf4s.Logger
 
+import java.lang.CharSequence
+import scala.concurrent.duration._
+
 private[destination] final class HANADestination[F[_]: ConcurrentEffect: MonadResourceErr: Timer](
     writeMode: WriteMode,
-    xa: Transactor[F],
+    xar: Resource[F, Transactor[F]],
```
Comment (wemrysi) [inline destination/src/main/scala/quasar/plugin/hana/destination/HANADestination.scala:44]: What required this to become a `Resource`?

File: destination/src/main/scala/quasar/plugin/hana/destination/TempTableFlow.scala [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+          val indexColumn =
+            if (idColumn.map(_.name) =!= filterColumn.map(_.name))
+              idColumn
+            else
+              none[Column[_]]
+
+          val prepare = writeMode match {
+            case WriteMode.Create =>
+              createTgt >>
+              idColumn.traverse_(index(tgtFragment, _)) >>
```
Comment (wemrysi) [inline destination/src/main/scala/quasar/plugin/hana/destination/TempTableFlow.scala:248]: Should this be `filterColumn`?

File: destination/src/main/scala/quasar/plugin/hana/destination/TempTableFlow.scala [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+              append
+            case WriteMode.Replace =>
+              dropTgtIfExists >>
+              rename >>
+              create
+            case WriteMode.Truncate =>
```
Comment (wemrysi) [inline destination/src/main/scala/quasar/plugin/hana/destination/TempTableFlow.scala:254]: Do we need to ensure an index on the `filterColumn` in the `Truncate` and `Append` cases?

File: destination/src/main/scala/quasar/plugin/hana/destination/HANADestination.scala (hunk 1 above)
Comment (cryogenian) [reply destination/src/main/scala/quasar/plugin/hana/destination/HANADestination.scala:44]: Hana driver throws an error with incorrect auth, customer needs to be able to create destination, do something with their db and run push after that.

File: destination/src/main/scala/quasar/plugin/hana/destination/TempTableFlow.scala (hunk 2 above)
Comment (cryogenian) [reply destination/src/main/scala/quasar/plugin/hana/destination/TempTableFlow.scala:248]: Nope, this should be `idColumn`. `filterColumn` is `None` for append sinks, but we might want to have index on target table.

File: destination/src/main/scala/quasar/plugin/hana/destination/TempTableFlow.scala (hunk 3 above)
Comment (cryogenian) [reply destination/src/main/scala/quasar/plugin/hana/destination/TempTableFlow.scala:254]: Right! This should have an index on `idColumn` as `WriteMode.Create` and `WriteMode.Replace` should have `indexColumn`. I'll update sql-server and postgres too, thank you

**Comment by cryogenian** [comment]
https://github.com/precog/quasar-plugin-sql-server/pull/72
https://github.com/precog/quasar-destination-postgres/pull/281

Snowflake doesn't use indices and it seems that teradata works as needed (due to lack of `CREATE INDEX` statement)

File: destination/src/main/scala/quasar/plugin/hana/destination/TempTableFlow.scala [hunk 4]
```diff
@@ -236,31 +236,37 @@ object TempTableFlow {
...
             case WriteMode.Create =>
               createTgt >>
               idColumn.traverse_(index(tgtFragment, _)) >>
               append
             case WriteMode.Replace =>
               dropTgtIfExists >>
               rename >>
-              create
+              create >>
+              indexColumn.traverse_(index(tgtFragment, _))
```
Comment (wemrysi) [inline destination/src/main/scala/quasar/plugin/hana/destination/TempTableFlow.scala:259]: Can we first create the index on the temp table, prior to renaming? I'm just wondering if index creation has any impact on the table performance, if so might as well wait until it is done before renaming.

**Review by wemrysi — APPROVED** [review]

File: destination/src/main/scala/quasar/plugin/hana/destination/TempTableFlow.scala (hunk 4 above)
Comment (cryogenian) [reply destination/src/main/scala/quasar/plugin/hana/destination/TempTableFlow.scala:259]: Like swap 258 and 259 lines, and use `tmpFragment`? Sure 👍

Comment (cryogenian) [reply destination/src/main/scala/quasar/plugin/hana/destination/TempTableFlow.scala:259]: Or move it to `create`?

Comment (wemrysi) [reply destination/src/main/scala/quasar/plugin/hana/destination/TempTableFlow.scala:259]: I was thinking instead index the temp table, prior to renaming. Looking at some of the other dbs, it probably isn't necessary as I've only seen index creation affect writes, so even if it does result in some locking it probably won't affect reads.

Comment (cryogenian) [reply destination/src/main/scala/quasar/plugin/hana/destination/TempTableFlow.scala:259]: Ok, so, I'll leave as is. We can return to it later if performance won't be OK?

**Pull request merged**